addon/components/frost-guage.js

```diff
@@ -0,0 +1,180 @@
+
+function getPercent (max, min, value) {
+  let valInt = parseInt(value, 10)
+  let maxInt = parseInt(max, 10)
+  let minInt = parseInt(min, 10)
+  let val = Math.min(Math.max(minInt, valInt), maxInt)
+  return ((val) / max)
+}
+
+export default Component.extend({
+  layout,
+  attributeBindings: ['style'],
+
+  propTypes: {
+    // options
+    elementPrefix: PropTypes.string,
+    min: PropTypes.number,
+    max: PropTypes.number,
+    direction: PropTypes.oneOf(['clockwise', 'counterclockwise']),
+    orientation: PropTypes.number,
+    data: PropTypes.arrayOf(PropTypes.object).isRequired,
+    size: PropTypes.number,
+    text: PropTypes.string
+  },
+
+  getDefaultProps () {
+    const elementPrefix = 'frost-guage'
+
+    return {
+      elementPrefix: elementPrefix,
+      classNames: [elementPrefix],
+      min: 0,
+      max: 100,
+      direction: 'clockwise',
+      // will move scale to propTypes once it's required. Consider it unsupported for now.
+      scale: 0,
+      orientation: 0,
+      negative: '#000000',
+      padding: 10
+    }
+  },
+
+  @readOnly
+  @computed('direction')
+  numericalDirection (direction) {
+    return (direction === 'counterclockwise') ? -1 : 1
+  },
+
+  @readOnly
+  @computed('size')
+  style (size) {
+    return Ember.String.htmlSafe('width: ' + size + 'px; height: ' + size + 'px;')
+  },
+
+  @readOnly
+  @computed('orientation', 'numericalDirection')
+  transform (orientation, numericalDirection) {
+    const d = parseInt(numericalDirection, 10)
+    const s = parseInt(this.get('size'), 10)
+    const o = orientation
+    const c = this.get('center')
+    const rotation = `rotate(${o - 90} ${c} ${c}`
+    const scale = `scale(${numericalDirection}, 1)`
+    let translate = ''
+    if (d === -1) {
+      translate = `translate(${s}, 0)`
+    }
+    return `${translate} ${scale} ${rotation})`
+  },
+
+  @readOnly
+  @computed('size')
+  center (size) {
+    return size / 2
+  },
+
+  @readOnly
+  @computed('size')
+  radius (size) {
+    return (parseInt(size, 10) - this.get('padding') * 2) / 2
+  },
+
+  @readOnly
+  @computed('diameter')
+  circumference (diameter) {
+    return diameter * Math.PI
+  },
+
+  @readOnly
+  @computed('radius')
+  diameter (radius) {
+    return radius * 2
+  },
+
+  @readOnly
+  @computed('orientation', 'numericalDirection')
+  arcs (orientation, numericalDirection) {
+    let transformList = []
+    let rotationOffset = 0
+
+    this.get('data').forEach((data) => {
+      let percent = getPercent(this.get('max'), this.get('min'), data.value)
+      let arcValue = (1 - percent) * this.get('circumference')
+      let transform = this.getTransform(rotationOffset)
+      let object = Ember.Object.create({
+        transform: transform,
+        offset: arcValue,
+        color: data.color,
+        label: data.label,
+        classes: data.classes
+      })
+      transformList.push(object)
+      rotationOffset += Math.floor(360 * percent)
```

Comment:
In some other cases I would agree, but if you're calculating degrees 360 isn't a magic number imo, nor is 1 a magic number when calculating a percentage, but that's just my take